src/models/order.js

```diff
@@ -0,0 +1,31 @@
+import Backbone from 'backbone';
+
+const Order = Backbone.Model.extend({
+  initialize(params) {
+    this.buy = params.buy,
+    this.currentQuote = params.currentQuote,
+    this.currentPrice = params.currentPrice,
+    this.targetPrice = params.targetPrice
+  },
+
+  validate(attributes) {
+    const error = {};
+    if (this.buy && this.targetPrice >= this.currentPrice) {
+      error.targetPrice = ['Price is higher than market, aim lower!'];
+    }
+    else if (!this.buy && this.targetPrice <= this.currentPrice) {
+      error.targetPrice = ['Price is lower than market, aim higher!'];
+    }
+    else if (!attributes.targetPrice) {
```

Comment:
You also need (based on your tests) to check for numeric value for price, boolean value for `buy` and maybe check `if (!attributes.targetPrice)` first.